extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -153,15 +152,16 @@ public Uni<? extends TenantConfigContext> get() {
     }
 
     private Uni<TenantConfigContext> initializeStaticTenantIfContextNotReady(TenantConfigContext tenantContext) {
-        if (tenantContext != null && !tenantContext.ready) {
+        requireNonNull(tenantContext, "tenantContext must never be null");
```

Comment:
This is private method, if it must never be null, make sure you don't call it with null. If this "tenantContext must never be null" is thrown, it hides where the actual issue is. For example with `resolveTenantConfigContext` we would throw `OIDCException`.

I can imagine that authentication doesn't make sense `tenantContext` and maybe we don't handle null everywhere. If tenant is missing, I'd say authentication should fail. If this shouldn't happen, which I honestly don't know, then I'd provide developer some log message or different exception messasge.

Reply:
Before this change, it would have silently returned a `null`-`Uni`, so I thought explicitly throwing an NPE would be better. WDYT?

Reply:
I think that you should look how is it used if you change it. If I read code correctly, one of the `resolveTenantConfigContext` now never receives null, therefore behavior changes as it used to throw `OIDCException`. I think it makes sense to add better null handling as you try, I just don't think that's it for reasons I mentioned before.

Reply:
My thinking was that you validate publicly accessible methods, if it is null here, fix the logic because it's illegal state. otherwise you will have to validate arguments in every method on every call.